tests/services/test_service.py

```diff
@@ -68,3 +69,38 @@ def test_simple_flow(app, service, identity_simple, input_data):
     # - search
     res = service.search(identity_simple, q=f"id:{id_}", size=25, page=1)
     assert res.total == 0
+
+
+def test_sorting_in_search(app, service, identity_simple, input_data):
+    """Search a record."""
+    idx = 'records-record-v1.0.0'
+
+    # Create an item
+    first_item = service.create(identity_simple, input_data)
+
+    input_data["metadata"]["title"] = "Another Test"
+    second_item = service.create(identity_simple, input_data)
+
+    # TODO: Should this be part of the service? we don't know the index easily
+    current_search.flush_and_refresh(idx)
+
+    # Search no query and no sort should sort by the most recent created record
+    res = service.search(identity_simple)
+
+    assert res.total == 2
+    assert list(res.hits)[0] == second_item.data
+
+    # Search with query and no sort should sort by the bestmatch record
+    res = service.search(identity_simple, q="title:Test")
+
+    assert res.total == 2
+    assert list(res.hits)[0] == first_item.data
+
+    # Search with existing sort parameter
+    res = service.search(identity_simple, sort="mostrecent")
+
+    assert res.total == 2
+    assert list(res.hits)[0] == second_item.data
+
+    # Search with non existing sort parameter
+    pytest.raises(ValidationError, service.search, identity_simple, sort="foo")
```

Comment:
Perhaps I am biased, but would simply copy *as-is* : https://github.com/inveniosoftware/invenio-records-resources/blob/master/tests/test_sorting.py 

I specifically made these tests at the endpoint level, because I was expecting the internals to change, but the ouput to stay the same from Lars+Alex's work. If I may say so myself (:wink: ), this is the perfect way to make sure we kept the API guarantees we provide through change to the code base (without having to rewrite tests either).

Reply:
The service is a high-level programmatic API that shouldn't be changing and breaking changes needs to be caught. I also prefer that each layer is well-tested on the features they provide - in this case it's the service that provides the sorting feature, while the resource "just" provide parameter parsing as this makes catching bugs easier - if the test is too high-level you don't know where in the stack you have too look. 
Invenio-RDM-Records should add the high-level tests that goes through the entire application stack and ensures that RDM stays with a stable API.

Reply:
I'll concede the addition of tests at the service level to ensure the programmatic interface of the service. But I would keep the endpoint level tests (enabled a number of them already), because records-resources does provide the core resource + service functionality. SImilar tests in rdm-records are fine, but if they break the fix is not in records-resources; whereas, tests here are great because if they break, the fix is in the same repository.